app/models/payout.rb

```diff
@@ -37,6 +37,10 @@ class Payout < ApplicationRecord
 	validate  :nonprofit_must_have_identity_verified, on: :create
 	validate  :bank_account_must_be_confirmed, on: :create
 
+	delegate :currency, to: :nonprofit
+
+	as_money :net_amount
```

Comment:
Good question! 

`as_money` accepts one or more arguments of method names (as symbols). These should be methods that return an amount of money in cents as an integer. For each of those of those method_names, it creates a new method named `<:method_name>_as_money`. So in this use-case, it would create a method called `net_amount_as_money`. These methods return an object of type `Amount`. Amount consists of two attributes, `:cents` and `:currency`. The cents comes from the original method and the currency comes from the `:currency` attribute on this instance. In most cases, we're delegating that currency up to Nonprofit.

Why does the `Amount` type exist? Houdini is intended to support multiple currencies. The idea is that you can pass around an object which corresponds to some amount of money and know:

a. what currency that amount is in, and
b. when you're passing something around between methods you know that it corresponds to some amount of money, not just some random number that could correspond to almost anything.

It's like the difference between passing around a `Time` object or the number of seconds since the [Unix epoch](https://en.wikipedia.org/wiki/Unix_time). Internally, a Time object holds the number of seconds since the Unix epoch. We pass them around as higher level objects which indicate what they do and what kind of operations we can perform on them. In Ruby,  we have `Time`,`TimeWithZone` and `Date` which all are different types of higher level objects to do so. Does that make sense?

As reference, the  `as_money` method is defined here:  https://github.com/wwahammy/houdini/blob/5e4e76a793fbd2b0ca4a8c0e2d1a86ea13a39bb2/app/models/concerns/model/as_moneyable.rb